interface.py

```diff
@@ -0,0 +1,621 @@
+    def replace(self, x, y):
+        global selected_piece
+        global pieces_path
+        global home_path
+        if (self.board[x][y] == selected_piece or selected_piece == "") and self.board[
+            x
+        ][y] != "":
+            self.content[x][y].config(image="")
+            self.board[x][y] = ""
+        elif selected_piece != "":
+            self.board[x][y] = selected_piece
+            image = self.getImage(selected_piece)
+            self.content[x][y].config(image=image)
+            self.content[x][y].image = image
+        self.main.fen_placement = self.get_fen()
+        self.main.display_fen()
+
+    def get_size(self, num):
+        global pieces_sizes
+        global pieces_padding
+        if num <= pieces_sizes[0] + pieces_padding:
+            return pieces_sizes[0]
+        else:
+            for i in pieces_sizes[1:]:
+                if num < i + pieces_padding:
+                    return i
+        return pieces_sizes[-1]
+
+    def getImage(self, piece):
+        global pieces_name
+        global home_path
+        global pieces_path
+        size = self.get_size(
+            (self.winfo_width() * (1 - 2 * self.border_proportion)) / 8
+        )
+        image = Image.open(
+            home_path
+            + "\\img\\"
+            + str(size)
+            + pieces_name
+            + pieces_path[piece]
+            + ".png"
+        )
+        return ImageTk.PhotoImage(image)
+
+    def get_fen(self):
+        fen = ""
+        space = 0
+        for y in range(8):
+            space = 0
+            for x in range(8):
+                if self.board[x][y] == "":
+                    space += 1
+                    continue
+                if space > 0:
+                    fen += str(space)
+                fen += self.board[x][y]
+                space = 0
+            if space > 0:
+                fen += str(space)
+            fen += "/"
+        return fen[:-1]
+
+    def set_board(self, fen):
+        for y, i in enumerate(fen.split("/")):
+            x = 0
+            for j in i:
+                if j.isnumeric():
+                    for k in range(int(j)):
+                        self.content[x][y].config(image="")
+                        self.board[x][y] = ""
+                        x += 1
+                else:
+                    self.board[x][y] = j
+                    image = self.getImage(j)
+                    self.content[x][y].config(image=image)
+                    self.content[x][y].image = image
+                    x += 1
+
+
+class Options(Frame):
+    def __init__(self, master, option_width):
+        self.main = master.main
+        Frame.__init__(self, master, bg=self.main.color_palette[5], width=option_width)
+        master.rowconfigure(0, weight=1)
+        master.columnconfigure(0, weight=1)
+        self._create_widgets(option_width)
+        self.last_selected_button = (0, 0)
+
+    def _create_widgets(self, option_width):
+        global pieces_name

```

Comment:
The "global" keyword is a code smell. If you just want to use constants, you may do that without global. And if you want to modify these variables, that means you're storing state in global variables, and that's a bad idea.